dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,225 @@
+|---------------------|---------------------------|--------------------------------------|
+| `hexRangeDistances` | `gridDiskDistancesUnsafe` | NONE                                 |
+| `_kRingInternal`    | `gridDiskDistancesSafe`   | NONE                                 |
+| `kRingDistances`    | `gridDiskDistances`       | `hexRangeDistances`,`_kRingInternal` |
+
+**Note**: The distances array is *optional* for `hexRangeDistances`, but *required* for the other two functions.
+
+#### Filled-In Disk Without Distances
+
+| Current name |              Proposed name               |                      Calls                       |
+|--------------|------------------------------------------|--------------------------------------------------|
+| `hexRange`   | `gridDiskUnsafe`                         | `hexRangeDistances`, does not allocate distances |
+| *DNE*        | `gridDiskSafe`                           |                                                  |
+| `kRing`      | `gridDisk`                               | `kRingDistances`, allocates and drops distances  |
+| `hexRanges`  | `gridDiskMultiUnsafe`, `gridDisksUnsafe` | N x `hexRange`                                   |
+
+- **TODO**: Do we *really* want to keep `hexRanges` in the API? It sounds like it currently has applications, but how
+  hard would it really be for a user to reproduce this functionality by just calling `hexRange` multiple times, and moving
+  the pointer before each call? It is also weird that we only provide the "unsafe" version of this function in the API.
+- **Idea**: leave it out initially, and see if anyone complains? It would be very easy to add back in.
+
+#### Hollow Ring
+
+| Current name |  Proposed name   | Calls |
+|--------------|------------------|-------|
+| `hexRing`    | `gridRingUnsafe` | NONE  |
+| *DNE*        | `gridRingSafe`   |       |
+| *DNE*        | `gridRing`       |       |
+
+
+### H3 Edge Types
+
+Instead of `UnidirectionalEdge`, use the term `DirectedEdge`.
+
+For a future undirected edge mode, use the term `Edge`.
+
+|                  Current name                 |        Proposed name         |
+|-----------------------------------------------|------------------------------|
+| `h3UnidirectionalEdgeIsValid`                 | `isValidDirectedEdge`        |
+| `getH3UnidirectionalEdge`                     | `cellsToDirectedEdge`        |
+| `getH3IndexesFromUnidirectionalEdge`          | `directedEdgeToCells`        |
+| `getH3UnidirectionalEdgesFromHexagon`         | `originToDirectedEdges`      |
+| *DNE*                                         | `destinationToDirectedEdges` |
+| `getH3UnidirectionalEdgeBoundary`             | `directedEdgeToLine`         |
+| `getOriginH3IndexFromUnidirectionalEdge`      | `getDirectedEdgeOrigin`      |
+| `getDestinationH3IndexFromUnidirectionalEdge` | `getDirectedEdgeDestination` |
+
+todo: happy with `directedEdgeToLine`?
```

Comment:
Are we happy with the name `directedEdgeToLine`? Is it clear that it would correspond to a GeoJSON `LineString`?

Reply:
Would `Polyline` make more sense here? That might align a bit better with `Polygon`

Reply:
What does the `poly` mean in `Polyline`? If anything, this is a line segment and so might be called something like `UniLine`.

In GeoJSON, there's no Polyline, just `LineString`. If we did call it `Polyline`, I wonder if users would confuse it with the GeoJSON `MultiLineString`. Thoughts?

Reply:
Ahh, I think I misunderstood what this function does. I was thinking it was representing a line as a pari of points from the lat/lng of the origin to the lat/lng of the destination.

Instead, it is representing the polygon edge(s) that are shared by the cells. (That's where the `Poly` comes from!)

Does this raise a question about how we might need to be careful about using the term "edge". There are "edges" of each polygon cell, and there are "edges" in the sense of the grid, connecting two adjacent cells. Depending on which one you're thinking of, the representations would be perpendicular.

Reply:
Also, this makes me wonder if we should reconsider the `Geo*` struct names to align with the changes

```
| `geoToH3`         | `pointToCell`   |
| `h3ToGeo`         | `cellToPoint`   |
| `h3ToGeoBoundary` | `cellToPolygon` |
```

That is,

- `GeoCoord`
- `GeoBoundary`
- `Geofence`
- `GeoPolygon`
- `GeoMultiPolygon`
- `LinkedGeoCoord`
- `LinkedGeoLoop`
- `LinkedGeoPolygon`
- `CoordIJ`

Reply:
I think it would be a good idea to align the internal data structure names with the API. Should be a straightforward find/replace.

`Polyline` is used in some other contexts (e.g. SVG) to describe a line with multiple segments, which may be true for a directed edge if it has a distortion vertex.

Reply:
I took a stab at this with the latest diff. Let me know how it looks!